Documentation/api_v3.md

```diff
@@ -0,0 +1,65 @@
+# etcd3 API
+
+TODO: API doc
+
+## KV API Guarantees
+
+etcd is a consistent and durable key value store with mini-transaction(TODO: link to txn doc when we have it) support. The key value store is exposed through the KV APIs. etcd tries to ensure the strongest consistency and durability guarantees for a distributed system. This specification enumerates the KV API guarantees made by etcd.
+
+### APIs to consider
+
+- Read APIs
+    - range
+    - watch
+- Write APIs
+    - put
+    - delete
+- Combination (read-modify-write) APIs
+    - txn
+
+### etcd Specific Definitions
+
+- operation completed
+    - An etcd operation is considered complete when it is committed through consensus, and therefore “executed” -- permanently stored -- by the etcd storage engine. The client knows an operation is completed when it receives a response from the etcd server. Note that the client may be uncertain about the status of an operation if it times out, or there is a network disruption between the client and the etcd member. etcd may also abort operations when there is a leader election. etcd does not send `abort` responses to  clients’ outstanding requests in this event.
```

Comment:
I would say we do not actually want to be that specific. Committed to permanent storage is the key idea, whatever that storage may be. 

(Curiousity: is there a lot of `fsync()`'ing or `bdflush()` kind of calls in etcd? That is, does it really concern itself with the disk or just do basic `write`s into the VFS?)